### PR TITLE
feat: multiple y axes

### LINKED ISSUE #827: [Feature]: Change Y-Axis orientation
### What problem does this feature solve?

I have a use case where I want to display the y-Axis on the right side of the chart. It seems like rechart [supports this](https://recharts.org/en-US/api/YAxis#orientation).

### What does the proposed API look like?

`orientation?: 'left' : 'right'` with the default being `'left'`. Maybe it would also be possible to just expose the rechart YAxis properties as `yAxisConfig` or similar?

### LINKED ISSUE #855: [Feature]: Multiple y-axes in AreaChart
### What problem does this feature solve?

I want to have two y-axes which have different domains. For example, y axis on the left side shows the amount of money while the one on the right side may shows the percentage.

It looks like recharts support this: https://codesandbox.io/p/sandbox/simple-line-chart-forked-c2h5cw. `yAxisId` and  `orientation` may be used.

Here is an example from another library: https://www.chartjs.org/docs/latest/samples/line/multi-axis.html

### What does the proposed API look like?

AreaChart may have optional `orientation` property which is an array of `left` and `right`.

Example: `orientation=["left", "right"]`

It means that the y axis of the first category is on the left side. The other one is on the right side. If `orientation` is provided, more than one `YAxis` (from recharts) can be used. If `orientation` is not provided as it is optional, the current behavior should not be changed.

This API also solves https://github.com/tremorlabs/tremor/issues/827.  `orientation=["right"]` may be passed in that case.

<hr />

In addition to `orientation`, `valueFormatter` property should be modified. Since the values on different axes are from different domains, they should be formatted differently. The `valueFormatter` function may have an optional second parameter: `index`.

```ts
function formatter(value, index) {
 switch (index) {
  case 0:
   return formatCurrency(value, locale, currency)
  case 1:
   return formatPercentage(value, locale)
 }
}
```

`valueFormatter={formatter}`

<hr />

I think neither adding `orientation` property nor modifying `valueFormatter` is a breaking change.

### PR DESCRIPTION
**Pull request by scaltunsoy**
**Description**

**Related issue(s)**

https://github.com/tremorlabs/tremor/issues/855
https://github.com/tremorlabs/tremor/issues/827

**What kind of change does this PR introduce?** (check at least one)

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New Feature (non-breaking change which adds functionality)
- [ ] New Feature (BREAKING CHANGE which adds functionality)
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [x] No

If yes, please describe the impact and migration path for existing applications:

**How has This been tested?**

Storybook. 4 new stories has been added in total.

**Screenshots (if appropriate):**

Multiple y axes:
image

Y axis on the right side:
image

**The PR fulfills these requirements:**

- [x] It's submitted to the `main` branch
- [x] When resolving a specific issue, it's referenced in the related issue section above
- [x] My change requires a change to the documentation. (Managed by Tremor Team)
- [x] I have added tests to cover my changes
- [x] Check the ["Allow edits from maintainers" option](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork) while creating your PR.
- [x] Add refs #XXX or fixes #XXX to the related issue section if your PR refers to or fixes an issue.
- [x] By contributing to Tremor, you confirm that you have read and agreed to Tremor's [CONTRIBUTING.md](https://github.com/tremorlabs/tremor/blob/main/CONTRIBUTING.md) guideline. You also agree that your contributions will be licensed under the [Apache License 2.0](https://github.com/tremorlabs/tremor/blob/main/License) license.

### CODE REVIEW

**Comment by leonlarsson** [comment]
This would absolutely come in handy

**Comment by jbergstroem** [comment]
My project requires two y axes as well -- anything blocking this to land?

File: src/components/chart-elements/AreaChart/AreaChart.tsx
Comment (Erasmus001) [inline src/components/chart-elements/AreaChart/AreaChart.tsx:1]: Uuum, i think you need to review this code `{categories.map((category, categoryIndex) => (
              <Area
                className={
                  getColorClassNames(
	@@ -396,6 +429,7 @@ const AreaChart = React.forwardRef<HTMLDivElement, AreaChartProps>((props, ref)
                animationDuration={animationDuration}
                stackId={stack ? "a" : undefined}
                connectNulls={connectNulls}
                yAxisId={orientations ? categoryIndex : undefined}
              />
            ))}`

i think you omitted the `key` in the component mapping, this is very crtitcal for React to know what to update and when to do so..

Comment (scaltunsoy) [reply src/components/chart-elements/AreaChart/AreaChart.tsx:1]: It is not omitted.

`key={categoryIndex}`

https://github.com/tremorlabs/tremor/pull/857/files#diff-9e1858f62ee598e304eddaa87b918d1d0788672840ac06754a3cc51a878a4818R218
https://github.com/tremorlabs/tremor/pull/857/files#diff-75f26982b036ec479134afab17386d622a8f0e0412378b0364b396d08ce5a528R214

**Comment by leonlarsson** [comment]
One feature I would love is to have multiple Y axes that are independent. Having checked the Storybook preview, it seems like one Y axis having a much higher number affects the other Y axis:


Ideally for me, I would have something like [this from MUI Charts](https://mui.com/x/react-charts/line-demo/#biaxiallinechart):
Here, one axis has a very high number, but it doesn't affect the display of the other axis.

**Comment by jbergstroem** [comment]
> One feature I would love is to have multiple Y axes that are independent. 

I think this is how it should work out of the box.

**Comment by scaltunsoy** [comment]
@leonlarsson 

image
image

I changed “Mar 21” data point of “Sales (blue line)” from 2000 to 15000 in Storybook.
The green line was not affected. It keeps its ratio.

**Comment by leonlarsson** [comment]
@scaltunsoy 

If you instead edit the other line ("Successful Payments"), it affects both series

**Comment by scaltunsoy** [comment]
@leonlarsson  

The blue line is not affected when I changed the green line. Do you have a specific setting other than `curveType="natural"`?
image

**Comment by leonlarsson** [comment]
@scaltunsoy Oh, sorry. I just noticed you're on AreaChart and I am looking at the LineChart. LineChart seems to have the behaviour I'm describing

**Comment by scaltunsoy** [comment]
Thank you @leonlarsson.

I think I fixed the issue. It seems like I forgot to pass `yAxisId` to the `Line` component.

**Comment by leonlarsson** [comment]
Great! Hoping to see this shipped :)

**Comment by gjohnsx** [comment]
hoping to see this implemented soon

**Comment by kalingwu** [comment]
we would love to have this feature made available!

**Comment by severinlandolt** [comment]
Hi @scaltunsoy accidentally deleted the beta branch, would you mind creating a new PR?

**Comment by scaltunsoy** [comment]
> Hi @scaltunsoy accidentally deleted the beta branch, would you mind creating a new PR?

Hi @severinlandolt. Here is the new PR: https://github.com/tremorlabs/tremor/pull/1028.

**Pull request closed without merging**